### PR TITLE
libtensorflow 1.14.0

### PR DESCRIPTION
**Pull request by bayandin**
Created with `brew bump-formula-pr`.

Also:
- use virtualenv instead of system python 
- added a patch to allow libtensorflow to be built on bazel 0.28.0. This patch a consolidation of the following patches (https://github.com/tensorflow/tensorflow/commit/c33055c44226499e99270ed05265ae3b960210c9, https://github.com/tensorflow/tensorflow/commit/040546fdefc5a111496dcd7be97b7d65594f91199 and https://github.com/tensorflow/tensorflow/commit/c3260eb5ab384e4d03e7e8c840712380492b26d9) + bumping version from 0.26.1 to 0.28.0
- added a patch with a fix for compatibility protobuf and bazel 0.28.0
- added new headers (to mirror https://github.com/tensorflow/tensorflow/commit/59b2a17c228be9cab5a8932417726e9afb931dc4)
- added .so and .dylib as per changelog: 
> Our libtensorflow tarball archives contain the libtensorflow library and two symlinks. MacOS .dylib libraries are the same, but match MacOS library naming requirements (i.e. libtensorflow.1.dylib):
> - libtensorflow.so.1.14.0, the main library
> - libtensorflow.so.1, symlinked to the main library
> - libtensorflow.so, symlinked to .so.1

### CODE REVIEW

**Comment by fxcoudert** [comment]
Have these issues been reported upstream? Please include a link

**Comment by bayandin** [comment]
@fxcoudert will do,

> - added a patch to allow libtensorflow to be built on bazel 0.26.1 (currently the latest available in homebrew)

This one, just a consolidation of https://github.com/tensorflow/tensorflow/commit/c33055c44226499e99270ed05265ae3b960210c9 https://github.com/tensorflow/tensorflow/commit/040546fdefc5a111496dcd7be97b7d65594f9119 and https://github.com/tensorflow/tensorflow/commit/c3260eb5ab384e4d03e7e8c840712380492b26d9

> - added new headers and .so and .dylib

This is just a reflection of https://github.com/tensorflow/tensorflow/commit/59b2a17c228be9cab5a8932417726e9afb931dc4 and this entry from the changelog: 
> - Our libtensorflow tarball archives contain the libtensorflow library and two symlinks. MacOS .dylib libraries are the same, but match MacOS library naming requirements (i.e. libtensorflow.1.dylib):
    - libtensorflow.so.1.14.0, the main library
    - libtensorflow.so.1, symlinked to the main library
    - libtensorflow.so, symlinked to .so.1


> added a workaround for ImportError: No module named builtins

And the last one I didn't sort out properly, there is a patch for tensorflow ci build scripts (https://github.com/tensorflow/tensorflow/pull/29233), which homebrew obviously doesn't use. Once I figure out the proper fix I'll create a patch or an issue for tensorflow and will update this PR with proper changes.

Thanks!

**Comment by bayandin** [comment]
Hey @fxcoudert,

I've updated the PR (and its description), could you please have a look if I need to add anything else?

Thanks!

File: Formula/libtensorflow.rb [hunk 1]
```diff
@@ -13,18 +15,24 @@ class Libtensorflow < Formula
...
+  # this patch a consolidation of the following patches
+  # - https://github.com/tensorflow/tensorflow/commit/c33055c44226499e99270ed05265ae3b960210c9 (from 0.25.2 to 0.25.3)
+  # - https://github.com/tensorflow/tensorflow/commit/040546fdefc5a111496dcd7be97b7d65594f9119 (from 0.25.3 to 0.26.0)
+  # - https://github.com/tensorflow/tensorflow/commit/c3260eb5ab384e4d03e7e8c840712380492b26d9 (from 0.26.0 to 0.26.1)
+  patch :DATA
 
   def install
+    venv_root = "#{buildpath}/venv"
+    venv = virtualenv_create(venv_root, "python2")
+    venv.pip_install("future>=0.17.1") # mimic changes from https://github.com/tensorflow/tensorflow/pull/29233 and https://github.com/tensorflow/tensorflow/pull/29138
```
Comment (SMillerDev) [inline Formula/libtensorflow.rb]: Can we use these as a patch now that they're merged?

Comment (bayandin) [reply Formula/libtensorflow.rb]: I don't think so, both https://github.com/tensorflow/tensorflow/pull/29233 and https://github.com/tensorflow/tensorflow/pull/29138 changes build scripts which homebrew formula doesn't use.

**Comment by bayandin** [comment]
@SMillerDev @fxcoudert is there anything else that I can do to get this merged?

**Comment by fxcoudert** [comment]
Can we compile with Python 3 instead?

**Comment by bayandin** [comment]
@fxcoudert I'd like to do in a separate PR, if you don't mind

**Comment by fxcoudert** [comment]
@bayandin I'd rather not introduce a dependency (even a build one) to switch it later. If this needs a non-system Python, it should be Python 3.

(It may also obliviate the need to install `future`, no?)

**Comment by bayandin** [comment]
@fxcoudert fair enough. I've updater PR

**Comment by fxcoudert** [comment]
```
==> ./configure
Extracting Bazel installation...
WARNING: ignoring _JAVA_OPTIONS in environment.
WARNING: --batch mode is deprecated. Please instead explicitly shut down your Bazel server using the command "bazel shutdown".
You have bazel 0.27.0-homebrew installed.
Please downgrade your bazel installation to version 0.26.1 or lower to build TensorFlow! To downgrade: download the installer for the old version (from https://github.com/bazelbuild/bazel/releases) then run the installer.
```

**Comment by bayandin** [comment]
@fxcoudert bumped bazel to 0.27.0 and added a patch to fix protobuf compatibility with the latest bazel.  Hope it'll be ok this time

**Comment by bayandin** [comment]
@fxcoudert is there anything else I can do?

**Comment by bayandin** [comment]
@fxcoudert can we merge it, please?

**Comment by fxcoudert** [comment]
Thanks @bayandin

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by bayandin** [comment]
@fxcoudert thanks! I also just updated bazel to 0.27.1 to make it build with the latest homebrew formula

**Comment by bayandin** [comment]
Bumped allowed bazel version to 0.28.0 (https://github.com/Homebrew/homebrew-core/pull/41896), tested locally — works fine.

**Pull request closed without merging**